context-ruby/spec/trunk_spec_helper.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+require 'rspec/core'
+require 'time'
+require 'context_ruby/context_ruby'
+
+def escape(str)
+  str.dump[1..-2]
+end
+
+module RSpec
+  module Core
+    # Example is the class that represents a test case
+    class Example
+      # keep the original method around so we can call it
+      alias set_exception_core set_exception
+      # RSpec uses the existance of an exception to determine if the test failed
+      # We need to override this to allow us to capture the exception and then
+      # decide if we want to fail the test or not
+      # trunk-ignore(rubocop/Naming/AccessorMethodName)
+      def set_exception(exception)
+        # TODO: this is where we'll need to override the result once the logic is ready
+        # trunk-ignore(rubocop/Lint/LiteralAsCondition)
+        if true
+          set_exception_core(exception)
+        else
+          # monitor the override in the metadata
+          metadata[:quarantined_exception] = exception
+          nil
+        end
+      end
```

Comment:
Handling quarantining in a later follow up. Leaving this logic here to show how it can be done.